core-rust/state-manager/src/transaction/round_update_transaction.rs

```diff
@@ -150,9 +150,13 @@ impl PreparedRoundUpdateTransactionV1 {
                     initial_proofs: btreeset!(AuthAddresses::validator_role()),
                     virtual_resources: BTreeSet::new(),
                 },
-                fee_payment: FeePayment::NoFee,
+                // TODO(during review): I copied that from Engine's "system transaction" handling
+                fee_payment: FeePayment {
+                    tip_percentage: 0,
+                    free_credit_in_xrd: DEFAULT_FREE_CREDIT_IN_XRD,
```

Comment:
I believe we should just use `0` here if we use `ExecutionConfig::for_system_transaction()` (which disables the costing module)

Reply:
huh, `0` looks good too, but to be sure, I'd first need an answer to https://github.com/radixdlt/babylon-node/pull/502/files/adffd10c8a6edd04edca1dce230c987ecbf45022#r1230033309

Reply:
Alright, as you said: this should be a system transaction and `0` 👍

Reply:
alright, so after "round update" started using `ExecutionConfig::for_system_transaction()` (as agreed in https://github.com/radixdlt/babylon-node/pull/502#discussion_r1230543028), then:
- the `0` here is giving me:
`thread '<unnamed>' panicked at 'round update rejected: RejectResult { error: ErrorBeforeFeeLoanRepaid(SystemModuleError(CostingError(FeeReserveError(LoanRepaymentFailed)))) }'`
- the `DEFAULT_FREE_CREDIT_IN_XRD` works fine.

Also, in the Engine, a system transaction uses `DEFAULT_FREE_CREDIT_IN_XRD` - see https://github.com/radixdlt/radixdlt-scrypto/blob/5547be8cf663e1a09df3758c0b400bfc25daaecc/transaction/src/model/v1/system_transaction.rs#L98-L116.
Is this expected behavior from the Engine's side?

_(either way, I am afraid we need `DEFAULT_FREE_CREDIT_IN_XRD`, at least at the moment)_

Reply:
That's a bit unexpected, but happy to keep `DEFAULT_FREE_CREDIT_IN_XRD` for now :man_shrugging:

Reply:
Yes, Lukas is right on this.

The costing module should've been disabled for `SystemTransactionV1` but the `enabled_module` flags aren't being respected in the code base (This has been fixed by https://github.com/radixdlt/radixdlt-scrypto/pull/1112).

For the time being, can we use `DEFAULT_FREE_CREDIT_IN_XRD` and add a `FIXME Change to zero`?